### PR TITLE
Corrections for `srclib coverage`

### PR DESCRIPTION
**Pull request by alexsaveliev**
The main goal of this pull request is to add changes needed to eliminate `srclib coverage` false negatives when ref points to nowhere
- Excluding "class", "super", "this" when emitting refs
- Emitting "class" refs to current class's type, "super" refs to current class's parent
- Forcing UnitType when emitting defs

### CODE REVIEW

**Comment by beyang** [comment]
Could we have `super` and `this` refer to their types instead for now?

**Comment by alexsaveliev** [comment]
> Could we have super and this refer to their types instead for now?
Done, however I think that `super` should refer to the parent's type

**Comment by beyang** [comment]
yes, I agree

File: src/main/java/com/sourcegraph/javagraph/TreeScanner.java [hunk 1]
```diff
@@ -405,4 +475,120 @@ public Void visitMemberSelect(MemberSelectTree node, Void p) {
...
+    /**
+     * @param name symbol to check
+     * @return true if name denotes "super" keyword
+     */
+    private boolean isSuper(CharSequence name) {
+        return "super".contentEquals(name);
+    }
+
+    /**
+     * @param node class node (class A extends B)
```
Comment (beyang) [inline src/main/java/com/sourcegraph/javagraph/TreeScanner.java]: Descriptive docstring, please.

File: src/main/java/com/sourcegraph/javagraph/TreeScanner.java [hunk 2]
```diff
@@ -405,4 +475,120 @@ public Void visitMemberSelect(MemberSelectTree node, Void p) {
...
+        JavaFileObject f = Origins.forElement(extendsElement);
+        URI defOrigin = null;
+        if (f != null) {
+            defOrigin = f.toUri();
+        }
+        return new DefKey(defOrigin, extendsElement.toString() + ":type");
+    }
+
+    /**
+     * @param node expression node (foo for foo.bar)
```
Comment (beyang) [inline src/main/java/com/sourcegraph/javagraph/TreeScanner.java]: Descriptive docstring, please.

File: src/main/java/com/sourcegraph/javagraph/TreeScanner.java [hunk 3]
```diff
@@ -405,4 +475,120 @@ public Void visitMemberSelect(MemberSelectTree node, Void p) {
...
+        JavaFileObject f = Origins.forElement(element);
+        URI defOrigin = null;
+        if (f != null) {
+            defOrigin = f.toUri();
+        }
+        return new DefKey(defOrigin, element.toString() + ":type");
+    }
+
+    /**
+     * @param node expression node ((parent of foo) for foo.bar)
```
Comment (beyang) [inline src/main/java/com/sourcegraph/javagraph/TreeScanner.java]: Descriptive docstring, please.

**Comment by beyang** [comment]
Other than docstring requests, LGTM. Can you merge after you've added those and then deploy the new image?

**Pull request merged**